docs/content/guides/dagster/local-to-production-tutorial.mdx

```diff
@@ -0,0 +1,420 @@
+We want to store the assets in a production Snowflake database, so we need to update the configuration for the `snowflake_io_manager`
+
+```python file=/guides/dagster/local_to_prod_tutorial/repository/repository_v3.py startafter=start endbefore=end
+# repository.py
+
+
+@repository
+def repo():
+    resource_defs = {
+        "hn_client": hn_api_client,
+        "io_manager": snowflake_io_manager.configured(
+            {
+                "account": "abc1234.us-east-1",
+                "user": "dev@company.com",
+                "password": "company_super_secret_password",
+                "database": "PRODUCTION",
+                "schema": "HACKER_NEWS",
+            }
+        ),
+    }
+
+    return [*with_resources([items, comments, stories], resource_defs=resource_defs)]
+```
+
+Here we run into an issue. The next time a developer wants to work on the assets, they will have to remember to change the configuration for the `snowflake_io_manager`. This leaves room for a developer to accidentally overwrite the production asset during local development.
+
+Instead we can set up the repository to attach resources to assets based on the environment:
+
+```python file=/guides/dagster/local_to_prod_tutorial/repository/repository_v4.py startafter=start endbefore=end
+# repository.py
+
+
+@repository
+def repo():
+    resource_defs = {
+        "local": {
+            "hn_client": hn_api_client,
+            "io_manager": snowflake_io_manager.configured(
+                {
+                    "account": "abc1234.us-east-1",
+                    "user": "me@company.com",
+                    "password": "my_super_secret_password",
+                    "database": "SANDBOX",
+                    "schema": "ALICE",
+                }
+            ),
+        },
+        "production": {
+            "hn_client": hn_api_client,
+            "io_manager": snowflake_io_manager.configured(
+                {
+                    "account": "abc1234.us-east-1",
+                    "user": "dev@company.com",
+                    "password": "company_super_secret_password",
+                    "database": "PRODUCTION",
+                    "schema": "HACKER_NEWS",
+                }
+            ),
+        },
+    }
+    deployment_name = os.environ.get("DAGSTER_DEPLOYMENT", "local")
+
+    return [
+        with_resources(
+            [items, comments, stories], resource_defs=resource_defs[deployment_name]
+        )
+    ]
+```
+
+Now, we can set the environment variable `DAGSTER_DEPLOYMENT=production` in our deployment and the correct resources will be applied to the assets.
+
+<Note>
+If you're using <a href="https://dagster.io/cloud">Dagster Cloud</a>, this environment variable will be set for you. Additionally, if you're using <a href="https://docs.dagster.cloud/guides/branch-deployments">Branch Deployments</a> <code>DAGSTER_DEPLOYMENT</code> will be automatically set to <code>VAR</code>.
+
+</Note>
+
+We still have some problems with this setup:
```

Comment:
This arc where we introduce varying resources and then take it away strikes me as kind of clunky. I'm left wondering "why is the guide showing me this need this thing that I don't need"?

Also, putting all the configuration in environment variables passes the buck to whomever is responsible for configuring the environment variables, and they might not have an understanding of the scheme we want to use for deciding where stuff goes in snowflake in different deployments.

What do you think about ending up on something like the following? The idea would be to have as much of the logic in Python as possible, but push out a few things to environment variables that can't be in Python, which is what I would recommend users in real life.
```
    resource_defs = {
        "local": {
            "hn_client": hn_api_client,
            "io_manager": snowflake_io_manager.configured(
                {
                    "account": "abc1234.us-east-1",
                    "user": {"env": "DEV_SNOWFLAKE_USER"},
                    "password": {"env": "DEV_SNOWFLAKE_PASSWORD"},
                    "database": "SANDBOX",
                    "schema": {"env": "DEV_SNOWFLAKE_USER"},
                }
            ),
        },
        "production": {
            "hn_client": hn_api_client,
            "io_manager": snowflake_io_manager.configured(
                {
                    "account": "abc1234.us-east-1",
                    "user": "system@company.com",
                    "password": {"env": "SYSTEM_SNOWFLAKE_PASSWORD"},
                    "database": "PRODUCTION",
                    "schema": "HACKER_NEWS",
                }
            ),
        },
    }
```